src/EditorFeatures/CSharpTest/UseAutoProperty/UseAutoPropertyTests.cs

```diff
@@ -39,6 +39,50 @@ int P
 }");
         }
 
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseAutoProperty)]
+        [WorkItem(28511, "https://github.com/dotnet/roslyn/issues/28511")]
+        public async Task TestMutableValueType1()
+        {
+            // Nullable<T> is a mutable value type. The diagnostic is not offered if the field is writable.
```

Comment:
That's super weird.  How is Nullable obervably mutable?  This actually seems like a regression for thsi scenario as the moment i have a nullable field, i can't use this feature?

Reply:
https://github.com/dotnet/coreclr/pull/15198

Reply:
➡️ I'll update the test to use a struct `MutableInt32` for resilience. I'll keep the `Nullable<T>` tests so the future behavior change, if any, is clear.

Reply:
That seems like a "them" problem, not an IDE problem :D.   I would still treat things like Nullable as being immutable since for 100% of our users it actually will be, and having this feature not show up is a worse cure :)   I'm fine for general, non-BCL, otherwise-mutable structs getting this treatment.

Reply:
`Nullable<T>` is observably mutable if `T` is observably mutable. I'm going to add tests for this case with a clear message on each test if the user finds themselves changing the expected outcome it means they are doing something wrong.

Reply:
> Nullable<T> is considered observably mutable if T is considered observably mutable

I don't see how that could be.  You can only get the inner value through a Property.  So it will always be copied out even if the inner T is mutable.  

Here's a simple example showing htat:

```c#
internal struct MStruct
{
    public int i;

    public void Mutate()
    {
        i++;
    }
}

internal class C
{
    static MStruct? m = new MStruct();

    private static void Main(string[] args)
    {
        m.Value.Mutate();
        Console.WriteLine(m.Value.i);
        Console.ReadLine();
```

Reply:
Oh, i see the repro case:

```c#
internal struct MStruct
{
    public int i;

    public override string ToString()
    {
        i++;
        return "";
    }
}

internal class C
{
    static MStruct? m = new MStruct();

    private static void Main(string[] args)
    {
        Console.WriteLine(m.ToString());
        Console.WriteLine(m.Value.i);
        Console.ReadLine();
```

IMO, this is so staggeringly unlikely htat it would be fine to still allow this.  We have far more likely situations that can occur with other refactorings/fixes, but we still allow because the 100% case for customers is unaffected.

Reply:
I'm also fine with your approach below of make this work for Nullable if T is not a mutable valuable type.  That way if i have an `int?` then i still get this feature.

Reply:
It's more likely to appear as a performance problem due to failure to cache a lazily-computed string. I like starting at the safe end and trying to find ways to address false positives without just ignoring the safety issue from the start.